ads/inabox/inabox-messaging-host.js

```diff
@@ -122,15 +122,24 @@ export class InaboxMessagingHost {
       return false;
     }
 
-    const iframe =
+    const adFrame =
         this.getFrameElement_(message.source, request['sentinel']);
-    if (!iframe) {
+    if (!adFrame) {
       dev().info(TAG, 'Ignored message from untrusted iframe:', message);
       return false;
     }
 
+    const allowedTypes = adFrame.iframe.dataset['ampAllowed'];
```

Comment:
Sorry I realized this was here in previous iterations but I'm slightly concerned that we're now mixing APIs where frames are added to the "whitelist" programmatically via global variable but allowed list is set via element attribute.  We should consider a single way to pass information so it does not become fragmented.

Reply:
I don't see how this is a problem (or another way to do this) considering that the whitelist themselves are an attribute of the frames, which is within the registered frames array. Another level down the hierarchy, so to speak.

Reply:
Nevermind, I saw your point. Will follow up later on.